app/grandchallenge/components/backends/base.py

```diff
@@ -48,8 +48,7 @@ def __init__(
         exec_image_repo_tag: str,
         memory_limit: int,
         time_limit: int,
-        requires_gpu: bool,
-        desired_gpu_type: GPUTypeChoices,
+        requires_gpu_type: GPUTypeChoices,
```

Comment:
Simplification of the API, we just allow an empty string (or anything False-y) to indicate not to attach a GPU.

Reply:
Ok, but isn't `requires_gpu_type` defined to be non-nullable on the model level now? So how would "" pass validation for a ComponentJob instance?

Reply:
It should pass as `""` is not `None`.

Reply:
We could set `blank=True`, but that is meaningless as the field is non-editable.

Reply:
Confirmed that `""` does pass validation.

Reply:
Ok, thanks. I always forget how CharFields handle "", and if it's not None, I would have thought that "" would only be allowed if it's included in the GPUTypeChoices. But anyway, thanks for checking!

Reply:
Good blog post on this topic (of course it's Adam Johnson): https://adamj.eu/tech/2020/01/22/djangos-field-choices-dont-constrain-your-data/